packages/libraries/federation-link-utils/src/index.ts

```diff
@@ -39,20 +38,27 @@ export function extractLinkImplementations(typeDefs: DocumentNode): {
   const linkByIdentity = Object.fromEntries(
     FederatedLink.fromTypedefs(typeDefs).map(l => [l.identity, l]),
   );
-  const supportsFederationV2 = linkByIdentity[FEDERATION_IDENTITY] !== undefined;
+  // Any schema with a `@link` directive present is considered federation 2
+  // although according to federation docs, schemas require linking specifically
+  // the federation 2.x spec. The reason for not being so picky is that supergraphs also
+  // use @link, but do not necessarily link to the federation 2.x spec.
+  const supportsFederationV2 = Object.keys(linkByIdentity).length > 0;
 
   return {
     resolveImportName: (identity, name) => {
       if (!supportsFederationV2) {
-        // @note identities dont matter for Federation v1. There are no links to reference.
+        // Identities dont matter for Federation v1. There are no links to reference.
+        // So return the name without the identity's namespace
         return name.startsWith('@') ? name.substring(1) : name;
       }
 
       const matchingLink = linkByIdentity[identity];
       if (!matchingLink) {
-        throw new Error(
```

Comment:
Instead of worrying about handling errors for this function, I think it makes more sense to return what the name would be assuming it was not imported, and nudge people to use `matchesImplementation` if they want to guarantee links are used. This is simpler and more intuitive for basic usage, but also more explicit for complicated cases.